servlet/src/main/java/io/undertow/servlet/attribute/ServletNameAttribute.java

```diff
@@ -43,7 +44,11 @@ private ServletNameAttribute() {
     @Override
     public String readAttribute(final HttpServerExchange exchange) {
         ServletRequestContext src = exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
-        return src.getCurrentServlet().getManagedServlet().getServletInfo().getName();
+        if (src != null) {
+            return src.getCurrentServlet().getManagedServlet().getServletInfo().getName();
+        } else {
+            return UNDEFINED;
```

Comment:
Could you explain the reasoning behind intentionally returning "No-Servlet" here? 

I was thinking that returning `null` would be sufficient, as it gets logged as `-` in the access log by default. This would also align with other ExchangeAttributes, which typically return `null` when a value isn't available.